src/components/FileUpload.tsx

```diff
@@ -1,41 +1,58 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 import { Input } from '@/components/ui/input';
 import { Label } from '@/components/ui/label';
 import Preview from '@/components/Preview';
 import { useToast } from '@/components/ui/use-toast';
+import axios from 'axios';
 
 const FileUpload = () => {
   const { toast } = useToast();
   const inputRef = React.useRef<HTMLInputElement>(null);
-  const [pdf, setPdf] = useState<string | null>(null);
+  const [filePath, setFilePath] = useState<string | null>(null);
+  const [pdf, setPdf] = useState<string | Blob>('');
 
   const clearInput = () => {
-    setPdf(null);
+    setFilePath(null);
     if (inputRef.current) {
       inputRef.current.value = '';
     }
   };
 
-  const onFileUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const file = e.target.files?.[0];
-    if (file) {
-      if (file.type === 'application/pdf') {
-        setPdf(URL.createObjectURL(file));
+  const onFileUpload = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const file = e.target.files?.[0];
+      if (file) {
+        if (file.type === 'application/pdf') {
+          setFilePath(URL.createObjectURL(file));
+          setPdf(file);
+        } else {
+          toast({
+            variant: 'destructive',
+            description: 'Incorrect file type, must be a pdf',
+          });
+          clearInput();
+        }
       } else {
-        toast({
-          variant: 'destructive',
-          description: 'Incorrect file type, must be a pdf',
-        });
         clearInput();
       }
-    } else {
-      clearInput();
-    }
-  };
+    },
+    [setFilePath, setPdf, toast],
+  );
 
-  const uploadFile = () => {
-    //Insert backend request here
-  };
+  const uploadFile = useCallback(async () => {
```

Comment:
nit: don't need async here since we are using `.then()`

Reply:
Fixed :)